vortex-array/src/compute/min_max.rs

```diff
@@ -0,0 +1,114 @@
+use vortex_dtype::Nullability::NonNullable;
+use vortex_error::{vortex_err, VortexError, VortexResult};
+use vortex_scalar::Scalar;
+
+use crate::stats::{Precision, Stat, Statistics};
+use crate::{Array, Encoding, IntoCanonical};
+
+pub type MinMaxResult = (Option<Scalar>, Option<Scalar>);
```

Comment:
Worth a `struct MinMax { pub min: ..., pub max: ... }`

Also.. can you document what the Option is for? I guess zero length? Shouldn't that be `Option<MinMax>` then?

Reply:
Its if either scalar cannot be computed

Reply:
I was actually thinking that we might add an option to the func to request only one of the two. values, then this would return only that one?

Reply:
Nahh, that's definitely a different compute function. Otherwise each array would have to internally switch kernels.